### PR TITLE
feat: possibility to change Repository Problems header in dependency dashboard issue

### PR DESCRIPTION
**Pull request by PhilipAbed**
## Changes

Added a config parameter to allow changing Repository Problems header, In the dependency dashboard issue.

## Context

The purpose is to be able to add "View Logs" and point to the actual logs of the run in the cloud hosted app.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

to test:
1) add these to the config.js:
 hashedBranchLength: 5,
  repoProblemsHeader: "press here [here](https://developer.mend.io/{{platform}}/{{repository}})",
2) run renovate

results:
image

in edit mode:
```
## Repository problems

press here [here](https://developer.mend.io/github/StinkyLord/scmany-nizo)

 - WARN: `hashedBranchLength` must allow for at least 6 characters hashing in addition to `branchPrefix`. Using 6 character hash instead.
```

### CODE REVIEW

File: lib/workers/repository/dependency-dashboard.ts [hunk 1]
```diff
@@ -174,8 +174,11 @@ function appendRepoProblems(config: RenovateConfig, issueBody: string): string {
       'repository problems'
     );
     newIssueBody += '## Repository problems\n\n';
-    newIssueBody +=
-      'These problems occurred while renovating this repository.\n\n';
+    if (config.repoProblemsHeader?.length) {
+      newIssueBody +=
+        template.compile(config.repoProblemsHeader, config) + '\n\n';
+    }
+    // newIssueBody += `${config.repoProblemsHeader}\n\n`;
```
Comment (viceice) [inline lib/workers/repository/dependency-dashboard.ts:181]: remove

File: lib/config/options/index.ts [hunk 2]
```diff
@@ -1977,6 +1977,13 @@ const options: RenovateOptions[] = [
     type: 'string',
     default: `This PR has been generated by [Renovate Bot](https://github.com/renovatebot/renovate).`,
   },
+  {
+    name: 'repoProblemsHeader',
```
Comment (viceice) [inline lib/config/options/index.ts:1981]: shouldn't we use a more general templates string dictionary, so it's later more extendable?

File: lib/workers/repository/dependency-dashboard.ts [hunk 3]
```diff
@@ -174,8 +174,11 @@ function appendRepoProblems(config: RenovateConfig, issueBody: string): string {
       'repository problems'
     );
     newIssueBody += '## Repository problems\n\n';
-    newIssueBody +=
-      'These problems occurred while renovating this repository.\n\n';
+    const repoProblemsHeader =
+      config.customizeDashboard?.['repoProblemsHeader'];
+    if (repoProblemsHeader) {
```
Comment (viceice) [inline lib/workers/repository/dependency-dashboard.ts:179]: can we use the old text in else block?

or should this text be optional?

@rarkins @secustor WDYT?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/workers/repository/dependency-dashboard.ts (hunk 3 above)
Comment (secustor) [reply lib/workers/repository/dependency-dashboard.ts:179]: I'm in favor for simply defaulting to empty string.

Comment (PhilipAbed) [reply lib/workers/repository/dependency-dashboard.ts]: i changed it, i agree it should be like that.

File: lib/workers/repository/dependency-dashboard.ts [hunk 4]
```diff
@@ -174,8 +174,11 @@ function appendRepoProblems(config: RenovateConfig, issueBody: string): string {
       'repository problems'
     );
     newIssueBody += '## Repository problems\n\n';
-    newIssueBody +=
-      'These problems occurred while renovating this repository.\n\n';
+    const repoProblemsHeader =
```
Comment (viceice) [inline lib/workers/repository/dependency-dashboard.ts:177]: please add a new test case to ensure the custom message is rendered.

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/workers/repository/dependency-dashboard.spec.ts [hunk 5]
```diff
@@ -641,6 +641,35 @@ describe('workers/repository/dependency-dashboard', () => {
...
+      ]);
+      config.dependencyDashboard = true;
+      config.customizeDashboard = {
+        repoProblemsHeader: 'platform is {{platform}}',
+      };
+
+      await dependencyDashboard.ensureDependencyDashboard(config, branches);
+
+      expect(platform.ensureIssue).toHaveBeenCalledTimes(1);
+      expect(platform.ensureIssue.mock.calls[0][0].body).toMatchSnapshot();
```
Comment (viceice) [inline lib/workers/repository/dependency-dashboard.spec.ts:670]: isn't a contains string value better that a snapshot?

Comment (PhilipAbed) [reply lib/workers/repository/dependency-dashboard.spec.ts:670]: i thought its more readable and easier to understand what the test does if you see the whole Repository Problems section

Comment (viceice) [reply lib/workers/repository/dependency-dashboard.spec.ts:670]: snapshots tend to be easily updated without behavior change notice, so an additional explicit check makes behavior changes more visible.

Comment (PhilipAbed) [reply lib/workers/repository/dependency-dashboard.spec.ts:670]: okey so ill do both? :D contains and snapshot? np..

**Review by viceice — DISMISSED** [review]

**Comment by viceice** [comment]
needs lint fix 😉

**Comment by PhilipAbed** [comment]
@viceice yes sorry, its ok now.

**Review by secustor — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**